### ISSUE TITLE
frequency of "woman"/"women" over time

### ISSUE THREAD

**Issue by Armand1** [Armand1/Women-in-the-BMJ#1]
I looked at the average frequency of the word "woman" or "women" in all papers by time. I used the dataset  "BMJ_171119_wordsbypapers_edited.csv" for this.  This is just word count data --- nothing to do with topics.

I first used grepl to get all words in which "woman" or "women" occurred (e.g., "chairwoman").  But it turned out that although there were many such words, most were OCR errors (e.g, "womenstudied" compounded together).  The vast majority were just "woman" and "women" by themselves. So I just looked at those. 

In an average paper, woman/en make up 0.25% of all words.  But this percentage increases from around 0.1% (incidence 0.001) to 0.5% (incidence 0.0055) in the course of 50 years.

![incidence women in articles](https://user-images.githubusercontent.com/8698079/69876893-a503e600-12b8-11ea-849b-df922fe69792.jpg)

I have modeled this in two ways.  First, as glm and also a gam, both based on based on the individual articles and pubdate rather than they yearly aggregates. The **pubdate** coefficient (odds ratio) of the glm is 1.022981 +-1.465534e-07 (95%CI), which is the factor by which "woman/en" appearing in a BMJ article increases per year.

**Comment by Armand1** [comment]
Now, previously, we worried that these kinds of results (albeit on the topics) were affected by the frequency of non-clinical papers.  They **should** be already excluded, but we can ask whether the rise of women is exceptional among a selection of clinical words. 

So I identified the 1000 most frequent clinical words in all articles (or rather, "women" + "woman" +1000 = 1002).  By "clinical" I mean a list of words from which I excluded the following kinds of words:  geography (e.g., London); abbreviations (e.g., "mg"); titles (e.g., "sir"), paper related (.e.g, "fig")

Then I did a GLM on each of them, as before, to see if the rise of women/an is exceptional The other 1000 words will act as a kind of control.  Many should decrease in frequency --- if they all increase, then we have a problem. 

**update** these are new results, based on GLMs with pubdates rather than year as the independent variable. So these results differ a little from before, but are better. 

Here are some of the most extreme that go up and down (note how they are non-linear). The fits here aren't meant to really accurately describe the detailed dynamics of these words, but only their overall trend.  


![Rplot01](https://user-images.githubusercontent.com/8698079/69882097-f915c700-12c6-11ea-8407-4d449f901b23.jpeg)

So it seems that our GLM fits are working well, and that we have clinically relevant words that are increasing and decreasing with frequency over time.  If we look at the odds ratio of all 1000 clinically relevant words we find that: (1) they are symmetrically distributed around 1; so lots go up and lots go down --- just as we hope; (2) the frequency of "woman/en" increases by a factor of
1.023 per year and that's in the 81st percentile ("woman/en"marked by a red line):

![Rplot](https://user-images.githubusercontent.com/8698079/69882308-ad175200-12c7-11ea-9f46-81506cfa2e68.jpeg)

So, not only is the frequency of woman/en increasing over time, it is increasing faster than 81% of other clinically relevant words.  

It is interesting to compare the incidence of "man/en" with "woman/en" on the same scale.  In 1948, they're both rather similar.  But where "man/en" has remained pretty constant since, "woman/en" has soared.  

![Rplot02](https://user-images.githubusercontent.com/8698079/69882615-b0f7a400-12c8-11ea-845e-84a9be008e6d.jpeg)

**Comment by evanhamulyak** [comment]
So, am I correct in stating that by using these generalized linear models based on the frequency of clinical words (_not_ topics) we demonstrate here the changing frequencies over time and more importantly that the frequency of 'women'/'woman' actually increases over 50 years from 0.1% to 0.5%, but this increase is also faster than 98% (AML: now 81%) of the other clinical words in our corpus.

**Comment by Armand1** [comment]
Exactly:  the idea here was simply to estimate the importance of women in a semi-independent way.  For example, this should ALSO capture the appearance of "woman/en" in articles that aren't about women's topics. For example, articles discussing, say, addiction might tend to mention women more often now than in the past --- even though addiction is hardly a women's health issue per se.   We can think of ways of looking at this idea more explicitly.

**Comment by Armand1** [comment]
I am increasingly convinced that paper length needs to be taken into account as a co-factor.  It varies hugely.  We saw  two spikes:  the first was around 1997:  that's due to OCR problems mentioned previously [here](https://github.com/Armand1/Women-in-the-BMJ/issues/2#issue-524965353) as something to look out for.  **I have now removed** those 1997 articles altogether.  Then, after Godlee becomes editor she evidently changes the format to have fewer, but longer, articles.  

![Rplot](https://user-images.githubusercontent.com/8698079/69914941-95fc6f80-1441-11ea-9fc8-0825bd7b480f.jpeg)

The way to deal with this is to have "mean_paper_length" as a co-factor in the GAMs and GLMs. Here is the GAM showing the effect of publication date and mean paper size on the probability that "woman/en" will appear in an article. You can see that it increases in both dimensions.


![Rplot01](https://user-images.githubusercontent.com/8698079/69915076-e32d1100-1442-11ea-8d77-b5525917ea9a.jpeg)

The pubdate effect seem much cleaner with the 1997 articles removed

A GLM shows the following P values

dec_pub_date                                     < 2e-16 ***
mean_paper_length                             2.18e-12 ***
dec_pub_date:mean_paper_length    1.22e-12 ***

So there is a very strong interaction between dec_pub_date and mean_paper_length.  This does not mean that woman/en (and the women's topics) don't increase with time:  just that a slightly more complex model is needed to see how they do.

**Comment by evanhamulyak** [comment]
I have a bit of difficulty interpreting this last graph. Am I correct in stating that:
1) Some sort of error occurred between 04/01/1997 and 14/02/1998, likely due to the OCR problems previously mentioned. Since we **_cannot_** adjust or correct for this, we should exclude these articles from the analyses and explain in the discussion section of the paper that we consider retrieved data not reliable in this period for this and this reason, therefore no conclusions can be drawn etc. (or something like that).
2) Whether it's changes in editors or changes by editors in BMJ format, the figure dec_pub_date:mean_paper_length confirmed your suspicion that paper length should be taken into consideration. Paper length is a confounding factor in our frequency results, but we **_can_** adjust for this, right?

**Comment by Armand1** [comment]
I wasn't totally clear.  So, this is the situation.  It turns out that paper length has an impact on whether or not "woman/en" will appear in a paper.  So we should model paper length as a factor, since doing so will make our estimate of the effect of pub date --- the thing we are interested in --- more accurate. 

In principle, having modeled paper length as a factor, we can keep the 1997 data.  But suddenly it becomes clear that the very thing that is wrong with them --- their excessive length --- has an effect on our trait of interest.  So I think we should exclude them. On balance, I think that's better than dealing with them statistically.   

Basically I don't want to present that surface (above) and then say: see the 1997 peak?  That's where our OCR went bad. I'd rather lose a year of data.

Obviously we can't just exclude the post 2000 papers just because they're long. So we have to deal with them statistically, but incorporating a "paper length" term in the model or, even a "pub date" x "paper length" (interaction) term.

**Comment by evanhamulyak** [comment]
Okay, that makes sense, thank you for elaborating and explaining! I have been trying to put this into writing (will add to overleaf later today), trying to find a hook. As (venous) thrombosis is our specific topic of interest, do you think we would be able to explore that association as discussed in a prior e-mail? We could set out the thrombosis observations to the general women's topics observations.

**Comment by Armand1** [comment]
If we incorporate another factor (paper length) into our model, to plot the response we should ideally have a 2 surface as above.  But that's a bit complicated to interpret --- and we're not interested in paper length. It just detracts from our story. 

So, instead, we can estimate the effect of pubdate on the probability of woman/en at a single paper length value: average paper length (1368 words).  That's like taking a slice through our 2d surface.

If we do that, then we can still clearly see that woman/en goes up.  But the curves look different from those estimated  [above](https://github.com/Armand1/Women-in-the-BMJ/issues/1#issue-524645287).  This is for two reasons.  A minor reason is that we've excluded 1997.  But a much more important reason is that here we have factored out the effect of paper length.

![Rplot02](https://user-images.githubusercontent.com/8698079/69915557-54bb8e00-1448-11ea-9237-3a922170f3e2.jpeg)

in this model, the annual rate of increase attributable to pub date alone is 1.03. Previously we estimated it as 1.023 --- so now it's a little higher.

It seems to me that this shows a clearer picture of woman/en rising in the 1980s, staying high, and then declining in late 2000s.  But any "Godlee effect" seems to disappear in the sense of we don't see a strong spike up near 2005.

This should be done with our topic data too.

**Comment by evanhamulyak** [comment]
Stumbled upon this, curious to hear what you think: https://www.theguardian.com/lifeandstyle/2019/nov/13/the-female-problem-male-bias-in-medical-trials

**Comment by Armand1** [comment]
Should we include paper-length as covariate? Now that we have gotten rid of the 1997 papers, I am not so sure.

[Previously ](https://github.com/Armand1/Women-in-the-BMJ/issues/4#issuecomment-561128484I )argued that the paper length and publication date * paper length coefficients for the aggregated women's topics were so small that we could neglect them --- even though they are statistically significant. Here I investigate whether that's also true for woman/en.

First the results of the models

**m1: lm(formula = cbind(N_present, N_absent) ~ dec_pub_date_stand, family = binomial,  data = a10)**

Coefficients:
                     Estimate Std. Error z value Pr(>|z|)    
(Intercept)        -5.8796183  0.0026616 -2209.1   **<2e-16** ***
dec_pub_date_stand  0.0227618  0.0001512   150.6   **<2e-16** ***

**m2: glm(formula = cbind(N_present, N_absent) ~ dec_pub_date _stand+mean_paper_length_stand, 
    family = binomial, data = a10)**
                     Estimate Std. Error z value Pr(>|z|)    
(Intercept)             -5.894e+00  2.841e-03 -2074.86   **<2e-16** ***
dec_pub_date_stand       2.417e-02  1.769e-04   136.60   **<2e-16** ***
mean_paper_length_stand -5.362e-05  3.402e-06   -15.76   **<2e-16** ***

**m3: glm(formula = cbind(N_present, N_absent) ~ dec_pub_date_stand * mean_paper_length_stand,  family = binomial, data = a10)**

Coefficients:
                                             Estimate Std. Error   z value Pr(>|z|)    
(Intercept)                                -5.865e+00  5.028e-03 -1166.476  **< 2e-16** ***
dec_pub_date_stand                          2.359e-02  1.933e-04   122.021  **< 2e-16** ***
mean_paper_length_stand                    -7.892e-06  7.258e-06    -1.087    **0.277**    
dec_pub_date_stand:mean_paper_length_stand -2.337e-06  3.290e-07    -7.103 **1.22e-12** ***

Again: mean_paper_length_stand and its interaction with pub date are really highly significant. But do they matter? Once again we look at the coefficients:

for pub date they're all basically the same.
m1: 1.023022855
m2: 1.023022855
m3: 1.023867012

And the paper length and its interaction with pubdate:
m3: paper_length: 0.999992108
m3: pub_date * paper_length: 0.999997663

Again: they're tiny compared to the main effect.  So I think we can neglect them.
